File: TestSample/go/world/world.go
package world

import "helloworld/person"

type World struct {
	people []person.Person
}

func New() *World {
	return &World{people: make([]person.Person, 0)}
}

func (w *World) Add(p person.Person) {
	w.people = append(w.people, p)
}

func (w *World) HelloAll() []string {
	out := make([]string, 0, len(w.people))
	for _, p := range w.people {
		out = append(out, p.Hello())
	}
	return out
}

File: TestSample/java/src/main/java/helloworld/World.java
package helloworld;

import java.util.ArrayList;
import java.util.List;

public final class World {
    private final List<Person> people = new ArrayList<>();

    public void add(Person person) {
        people.add(person);
    }

    public List<String> helloAll() {
        List<String> lines = new ArrayList<>();
        for (Person p : people) {
            lines.add(p.hello());
        }
        return lines;
    }
}
